solver_greedy_max.py

```diff
@@ -8,6 +8,9 @@
 from astropy.visualization import time_support
 time_support()
 
+from resource_mock import Resource
```

Comment:
Any imports should always be before function calls, so you should swap this with `time_support()`, as this is Python standard.